src/serde/read_cache_lookup.rs

```diff
@@ -16,9 +18,35 @@
 ///
 /// All hashes correspond to sha256 tree hashes.
 use std::collections::{HashMap, HashSet};
+use std::hash::{BuildHasher, Hasher};
 
 use super::bytes32::{hash_blob, hash_blobs, Bytes32};
 
+#[derive(Default, Clone, Copy)]
+pub struct IdentityHash(u64);
+
+impl Hasher for IdentityHash {
+    fn finish(&self) -> u64 {
+        self.0
+    }
+
+    fn write(&mut self, bytes: &[u8]) {
+        self.0 = u64::from_le_bytes(bytes[0..8].try_into().expect("expected 32 byte hashes"));
+    }
+
+    fn write_u64(&mut self, _i: u64) {
+        panic!("This hasher only takes bytes");
```

Comment:
I don't know if this is necessary. If you want to use a standalone hasher I tried a couple years back, you could steal it from https://github.com/richardkiss/clvm_rs/commit/50f6c94a02b909d28c5479453ff2d9d8e2800749#diff-b696b12532e34a5d93b4bf96927a9049a2ad3d7c04bfde786611f6a560622e2f ; I don't think it'd be much slower, if at all.

Reply:
this "hasher" depends on the input already having hash-properties. If anyone, for some reason, puts a `u64` into it, that assumption likely no longer holds. I think it's preferable to panic in that case.

That said, the one risk I see with this is that there's no salt, so an attacker can grind hashes that causes worst-case performance of the hash set/map.

Reply:
Possible solution; don't always take the first eight bytes, but randomly select an offset between 0 and 24 to start taking the bytes.

True, that only makes it 24 times as hard. But we could choose, say, five random offsets between 0 and 24 and eor them together. Then it's 24**5 times as hard.

Reply:
One attack I see would be, create a legit spend that has so many hash tree collisions that the hash tree gets O(n^2) performance. If we only have a 1/24 chance of having that happen, it won't be long before it gets included in a block by someone who chooses a failing offset, and then the spend is gone forever.

That's really the only attack I can think of.

Reply:
I've addressed this now